File: src/main/java/com/vaadin/demo/component/textarea/TextAreaAutoHeight.java
package com.vaadin.demo.component.textarea;

import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.textfield.TextArea;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.demo.domain.DataService;

@Route("text-area-auto-height")
public class TextAreaAutoHeight extends Div {

  public TextAreaAutoHeight() {
    String loremIpsum = DataService.getTemplates().getLoremIpsum();

    // tag::snippet[]
    TextArea textArea = new TextArea();
    textArea.setWidthFull();
    textArea.setLabel("Description");
    textArea.setValue(loremIpsum);
    add(textArea);
    // end::snippet[]
  }

  public static class Exporter extends DemoExporter<TextAreaAutoHeight> { // hidden-source-line
  } // hidden-source-line
}

File: src/main/frontend/demo/component/textarea/text-area-auto-height.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/text-area';
import { css, html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';
import templates from '../../../../src/main/resources/data/templates.json';

@customElement('text-area-auto-height')
export class Example extends LitElement {
  static override styles = css`
    vaadin-text-area {
      width: 100%;
    }
  `;

  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-text-area label="Description" value="${templates.loremIpsum}"></vaadin-text-area>
      <!-- end::snippet[] -->
    `;
  }
}
